src/metadata/EntityListenerMetadata.ts

```diff
@@ -27,7 +28,7 @@ export class EntityListenerMetadata {
      * Target class to which metadata is applied.
      * This can be different then entityMetadata.target in the case if listener is in the embedded.
      */
-    target: Function|string;
+    target: Function;
```

Comment:
why?

Reply:
don't forget about [entity schemas](https://typeorm.io/#/separating-entity-definition).

Reply:
not sure if they support entity listeners however. But we need to support them if they are not.

Reply:
looks like #2089 mentions lack of listeners for entity schema. I'll see about another PR to add listeners for entity schema